test/integration/targets/mongodb_shard/tasks/main.yml

```diff
@@ -0,0 +1,443 @@
+
+- name: Create config srv replicaset with module
+  mongodb_replicaset:
+    login_user: "{{ mongodb_admin_user }}"
+    login_password: "{{ mongodb_admin_password }}"
+    login_port: 4000
+    login_database: "admin"
+    replica_set: "{{ configsrv_replicaset }}"
+    validate: no
+    members:
+     - "localhost:4000"
+
+- name: Ensure host reaches primary before proceeding 3001
+  command: mongo admin --port 3001 "{{ remote_tmp_dir }}/tests/is_primary.js"
+
+- name: Ensure host reaches primary before proceeding 3004
+  command: mongo admin --port 3004 "{{ remote_tmp_dir }}/tests/is_primary.js"
+
+- name: Ensure host reaches primary before proceeding 4000
+  command: mongo admin --port 4000 "{{ remote_tmp_dir }}/tests/is_primary.js"
+
+- name: Add mongodb admin user to each shard 3.4+
+  mongodb_user:
+    login_host: localhost
+    login_port: "{{ item.port }}"
+    replica_set: "{{ item.rs }}"
+    database: admin
+    name: "{{ mongodb_admin_user }}"
+    password: "{{ mongodb_admin_password }}"
+    roles: ["root"]
+    state: present
+  register: mongo_admin_user
+  with_items:
+    - { "port": 3001, "rs": "{{ mongodb_replicaset1 }}" }
+    - { "port": 3004, "rs": "{{ mongodb_replicaset2 }}" }
+    - { "port": 4000, "rs": "{{ configsrv_replicaset }}" }
+  when: mongodb_version not in ["3.2", "4.0"] and test_mongo_auth == True
+
+- name: Add mongodb admin user to each shard 3.2 ^ 4.0
+  mongodb_user:
+    login_host: localhost
+    login_port: "{{ item.port }}"
+    replica_set: "{{ item.rs }}"
+    database: admin
+    name: "{{ mongodb_admin_user }}"
+    password: "{{ mongodb_admin_password }}"
+    roles: ["root"]
+    state: present
+  register: mongo_admin_user
+  with_items:
+    - { "port": 3001, "rs": "{{ mongodb_replicaset1 }}" }
+    - { "port": 3004, "rs": "{{ mongodb_replicaset2 }}" }
+  when: mongodb_version not in ["3.2", "4.0"] and test_mongo_auth == True
+
+# mongodb_user throws an error when creating a user on 3.2 (also on 4.0 with Ubuntu 18.04)
+# 'majority' is the only valid write concern when writing to config server replica sets
+- name: Copy create_user_root_3.2.js.j2 template to host
+  template:
+    src: create_user_root_3.2.js.j2
+    dest: /root/create_user_root_3.2.js
+  when: mongodb_version in ["3.2", "4.0"]
+
+- name: Copy script to host
+  template:
+    src: files/bash/ensure_primary.sh.j2
+    dest: /root/ensure_primary.sh
+
+- name: Execute script for 3001
+  script: /root/ensure_primary.sh 3001 0
+# We do this here because sometimes 3004 instance seems to be demoted
+- name: Execute script for 3004
+  script: /root/ensure_primary.sh 3004 0
+
+- name: Create admin user on 3.2 and 4.0 config replset
+  shell: mongo admin --port {{ item }} /root/create_user_root_3.2.js
+  with_items:
+    - 4000
+    - 3001
+    - 3004
+  when: mongodb_version in ["3.2", "4.0"]
+
+- name: Murder all mongod processes
+  shell: pkill mongod && sleep 10 || sleep 10;
```

Comment:
Same comment as earlier about waiting for the process to exit instead of sleeping.